android-components/components/browser/state/src/main/java/mozilla/components/browser/state/reducer/AwesomeBarStateReducer.kt

```diff
@@ -25,6 +25,9 @@ internal object AwesomeBarStateReducer {
             is AwesomeBarAction.EngagementFinished -> {
                 state.copy(awesomeBarState = AwesomeBarState())
             }
+
+            AwesomeBarAction.EngagementStarted ->
+                state.copy(awesomeBarState = state.awesomeBarState.copy(isActive = true))
```

Comment:
That's a good suggestion, I was also considering that, but after closer examination found out no way to achieve what we want based on the existing data. `VisibilityState` contains a list of providers and suggestions for the particular input, and until there is any input, it is empty. It makes the state where there is no search dialog fragment and where a search dialog fragment is present but has no suggestions indistinguishable. 

Another reason I would like to introduce a new parameter is transparency. `AwesomeBarState.visibilityState` was designed to hold the list of suggestions, not actually represent the visibility of the search bar itself. So, even if we could calculate it from `visibilityState`, any change to the behaviour of `visibilityState` would break the toolbar logic. But introducing a `AwesomebarState.isActive` in combination with `AwesomeBarAction.EngagementFinished` & `AwesomeBarAction.EngagementStarted` is very clear.

Reply:
Okay, I see there's no easy way to derive this with the existing properties. My initial thinking was that we don't usually expect to see a UI-specific state such as `isActive` in `BrowserState`. Hence, the attempt to try and derive this via extension functions instead. I think the best way to proceed here would be to add this state (`isAwesomeBarActive`) inside of our `AppState` (assuming this is needed globally) simply because we don't need this `isActive` within the AC `feature-` and `browser-` components therefore this state is specific to Fenix. @rocketsroger What do you think of this suggestion?

Reply:
@gabrielluong that sounds good to me.

Reply:
thanks for the input, guys

I change the name to `isSearchDialogVisible`. The `AwesomeBar` isn't what we actually depend upon, it's the thing that is showing suggestions and it just so happens that it gets active with the `SearchDialogFragment`. I was considering other names like 'isToolbarFocused' but can't say which is the best fit. Open to suggestions if anybody feels strong about it, but for now decided to stick with the one that looked general and representative the most.